flocker/node/agents/blockdevice.py

```diff
@@ -1395,6 +1418,61 @@ def _manifestation_from_volume(volume):
     return Manifestation(dataset=dataset, primary=True)
 
 
+def _ext4_fs_uuid(device_path):
+    """
+    Get the EXT4 filesystem UUID from the filesystem on the device at the given
+    path.
+
+    :param FilePath device_path: The path to the device on which the filesystem
+        resides.
+
+    :return: A ``UUID`` instance giving the UUID from the filesystem.
+    :rtype: ``UUID``
+    """
+    blkid = _blkid(
```

Comment:
:question: Apparently `blkid` uses a cache by default. Could it give us stale results?
```
man blkid
...
       -p     Switch to low-level superblock probing mode (bypassing the cache).

              Note that low-level probing also returns information about partition table type (PTTYPE tag) and partitions (PART_ENTRY_* tags).
```